packages/gatsby/src/redux/persist.ts

```diff
@@ -125,6 +151,20 @@ function prepareCacheFolder(
       )
     }
   }
+
+  if (pagesMap) {
```

Comment:
Is it possible for `pagesMap` to be nullable here? I understand the property is nullable from the typing pov, but at this point in the code I think the pages ought to exist, right?

Reply:
I think it should be at least empty Map, yes. To be fair - this is just copied part from https://github.com/gatsbyjs/gatsby/blob/f24a23d5b1f24ab33aec7a8ac882dbbccdf7d01e/packages/gatsby/src/redux/persist.ts#L113-L127 which also use same check. I guess this is mostly to apease TS gods. And nullable part is also copied from how `nodes` is typed in ICachedReduxState and reason I guess is that we set it to `undefined` temporaritly while we save `rest` part of the sate